go.mod

```diff
@@ -1,4 +1,4 @@
-module github.com/equinix/cloud-provider-equinix-metal
+module github.com/kubernetes-sigs/cloud-provider-equinix-metal
```

Comment:
Let's use [`sigs.k8s.io/cloud-provider-equinix-metal`](https://sigs.k8s.io/cloud-provider-equinix-metal) for the module name (and any Go module/import references to it), consistent with CAPP.

Reply:
Change has been made, can you resolve if ok?